app/models/subnet.rb

```diff
@@ -102,22 +120,40 @@ def <=>(other)
   end
 
   # Given an IP returns the subnet that contains that IP
-  # [+ip+] : "doted quad" string
+  # [+ip+] : IPv4 address
   # Returns : Subnet object or nil if not found
   def self.subnet_for(ip)
-    Subnet.all.each {|s| return s if s.contains? IPAddr.new(ip)}
-    nil
+    ip = IPAddr.new(ip)
+    Subnet.all.detect {|s| s.family == ip.family && s.contains?(ip)}
   end
 
   # Indicates whether the IP is within this subnet
-  # [+ip+] String: Contains 4 dotted decimal values
+  # [+ip+] String: IPv4 address
   # Returns Boolean: True if if ip is in this subnet
   def contains?(ip)
-    IPAddr.new("#{network}/#{mask}", Socket::AF_INET).include? IPAddr.new(ip, Socket::AF_INET)
+    ipaddr.include? IPAddr.new(ip, family)
+  end
+
+  def ipaddr
+    IPAddr.new("#{network}/#{mask}", family)
   end
 
   def cidr
+    return if mask.nil?
     IPAddr.new(mask).to_i.to_s(2).count("1")
+  rescue ArgumentError
```

Comment:
@dLobatog : On Ruby 1.9 ```IPAddr::InvalidAddressError``` is not defined. IPAddr throws an ArgumentError there. How should we proceed?